runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,421 @@
+		return ircClient;
+	}
+
+	//Send messages to twitch chat using command "t"
+	@Subscribe
+	private void onCommand(CommandExecuted executed)
+	{
+		if (!client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			return;
+		}
+
+		String[] args = executed.getArguments();
+		StringBuilder sb = new StringBuilder();
+		String command = executed.getCommand();
+
+		if (command.equalsIgnoreCase("t"))
+		{
+			for (String arg : args)
+			{
+				sb.append(arg + " ");
+			}
+
+			String message = sb.toString();
+			irc.sendMessage("#" + config.nickName(), message);
+			//Echos the message back in in-game chat
+			if (irc.getChannels().toString().contains("#" + config.nickName()))
+			{
+				//If the chat is in sub only mode and they are not a subscriber
+				if (!subscribed && subsOnly)
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=ffffff></col>[<col=9070ff>Twitch</col>] Notification: <col=3D2BAD>This channel is in Subscriber only mode!</col>", "");
+				}
+				//If chat is in emote only mode
+				else if (emoteOnly && !broadcaster && !mod)
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=000000></col>[<col=0000FF>Twitch</col>] Notification: <col=3D2BAD>This channel is in emote only mode!<br><col=3D2BAD>Function coming soon to a Runelite client near you.</col>", "Twitch");
+				}
+				else
+				{
+					sendMessage(message, badges + config.channelName());
+				}
+			}
+		}
+	}
+
+	private void sendMessage(String message, String name)
+	{
+		String oColor = Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF);
+		String tColor = Integer.toHexString(config.chatTColor().getRGB() & 0xFFFFFF);
+
+		if (isChatboxTransparent())
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+				"<col=ffffff></col>[<col=9070ff>Twitch</col>] " +  name + ": <col=" + tColor + ">" + message + "</col>", "");
+		}
+		else
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+				"<col=000000></col>[<col=0000FF>Twitch</col>] " +  name + ": <col=" + oColor + ">" + message + "</col>", "");
+		}
+	}
+
+	/**
+	 * Shut off current running irc client
+	 * @param c
+	 */
+	private void twitchShutdown(Client c)
+	{
+		try
+		{
+			c.shutdown();
+		}
+		catch (Exception e)
+		{
+			log.debug("Not connected to a chat yet.", e);
+		}
+	}
+
+	/**
+	 * Triggers when a message is sent to twitch chat
+	 * @param event
+	 */
+	@Handler
+	private void onMessage(ChannelMessageEvent event)
+	{
+		if (!client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			return;
+		}
+
+		String username = event.getActor().getUserString();
+		String badge = event.getOriginalMessages().get(0).getTag("badges").toString();
+
+		username = getBadges(badge, username);
+
+		String message = event.getMessage();
+		sendMessage(message, username);
+	}
+
+	/**
+	 * Triggers when a /me message is sent in twitch chat
+	 * Works the same as the onMessage method
+	 * @param event
+	 */
+	@Handler
+	private void onMe(ChannelCtcpEvent event)
```

Comment:
The vast majority of this logic is copy pasted from the above method, and should be abstracted out to its own method to reduce superfluous code.

Reply:
I'm not entirely sure how I should go about this. I made some changes but I don't think it was what you're looking for. I'm probably missing something stupidly obvious. I've committed the changes so please take a look

Reply:
If you compare the two methods, both of them perform

```java
String username = event.getActor().getUserString();
String badge = event.getOriginalMessages().get(0).getTag("badges").toString();
``` 
and
```java
messageBuilder(username, message, badge);
```

The only way they differ is how they construct the message, so if you add a new method, to which you pass
1) the event
2) a message

you can change your logic to be 
```java
String message = event.getMessage().replace("ACTION ", "");
someMethod(event, message)
```

```java
String message = event.getMessage();
someMethod(event, message)```

Reply:
So should I have overloaded methods?
i.e. 
`someMethod(ChannelCtcpEvent event, String message) { ... }`
`someMethod(ChannelMessageEvent event, String message) { ... }`
Just want to make sure I'm not missing something

Reply:
Hah, I didn't actually even notice that they were different - Two methods with near-identical logic, next to each other, where the input is called `event` in both scenarios despite being different objects.

Regardless both extend `ActorChannelMessageEventBase<User>` so you can just make the method take that, and pass both objects straight to it